# Les librairies nécessaires

In [10]:
import pandas as pd 
import seaborn as sns
import warnings 
import os
import sqlalchemy

from datetime import datetime
warnings.filterwarnings(action="ignore")

%pylab inline
%load_ext sql

plt.style.use('seaborn-darkgrid')
sns.set(font_scale=3)

%pylab is deprecated, use %matplotlib inline and import the required libraries.
Populating the interactive namespace from numpy and matplotlib
The sql extension is already loaded. To reload it, use:
  %reload_ext sql


# La connexion à la base de données PostgreSQL

**`postgresql://user:pass@host:port/dbname[?key=value&key=value...]`**<br>
**`engine = create_engine("postgresql://scott:tiger@localhost/test")`**<br>

In [11]:
%sql postgresql://

# L’opérateur égal à


>> SELECT [ ALL | DISTINCT ] <br>
>>> { [{ NOM_TABLE | ALIAS2 }.]* ,<br><br>
>>> [ [{NOM_TABLE |ALIAS2}.]COLONNE1 [AS] ALIAS1[,...]} 

>> FROM NOM_TABLE ALIAS2

>> WHERE PREDICAT<br>



In [12]:
%%sql 
SELECT * 
FROM CLIENTS 
WHERE PAYS='France'
FETCH NEXT 5 ROWS ONLY

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
5 rows affected.


[('DUMON', 'Du monde entier', '67, rue des Cinquante Otages', 'Nantes', '44000', 'France', '02.40.67.88.88', '02.40.67.89.89'),
 ('FOLIG', 'Folies gourmandes', '184, chaussée de Tournai', 'Lille', '59000', 'France', '03.20.16.10.16', '03.20.16.10.17'),
 ('FRANR', 'France restauration', '54, rue Royale', 'Nantes', '44000', 'France', '02.40.32.21.21', '02.40.32.21.20'),
 ('LACOR', "La corne d'abondance", "67, avenue de l'Europe", 'Versailles', '78000', 'France', '01.30.59.84.10', '01.30.59.85.11'),
 ('LAMAI', "La maison d'Asie", '1 rue Alsace-Lorraine', 'Toulouse', '31000', 'France', '05.61.77.61.10', '05.61.77.61.11')]

In [13]:
%%sql 
SELECT NOM, PRENOM, DATE_EMBAUCHE 
FROM EMPLOYES 
WHERE DATE_EMBAUCHE = '1991-04-03'

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
2 rows affected.


[('Maillard', 'Corinne', datetime.date(1991, 4, 3)),
 ('Thimoleon', 'Georges', datetime.date(1991, 4, 3))]

## IS NULL

In [14]:
%%sql 
SELECT NOM, PRENOM, SALAIRE, COMMISSION 
FROM EMPLOYES 
WHERE COMMISSION IS NULL
FETCH NEXT 5 ROWS ONLY

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
5 rows affected.


[('Giroux', 'Jean-Claude', Decimal('150000.00'), None),
 ('Fuller', 'Andrew', Decimal('96000.00'), None),
 ('Brasseur', 'Hervé', Decimal('147000.00'), None),
 ('Poupard', 'Claudette', Decimal('1800.00'), None),
 ('Maurer', 'Véronique', Decimal('1400.00'), None)]

## IS NOT NULL

In [15]:
%%sql 
SELECT NOM, PRENOM, SALAIRE, COMMISSION 
FROM EMPLOYES
WHERE COMMISSION IS NOT NULL
FETCH NEXT 5 ROWS ONLY

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
5 rows affected.


[('Buchanan', 'Steven', Decimal('13000.00'), Decimal('12940.00')),
 ('Leger', 'Pierre', Decimal('19000.00'), Decimal('11150.00')),
 ('Belin', 'Chantal', Decimal('10000.00'), Decimal('10640.00')),
 ('Chambaud', 'Axelle', Decimal('12000.00'), Decimal('11600.00')),
 ('Ragon', 'André', Decimal('13000.00'), Decimal('5980.00'))]

# Les opérateurs logiques

## Supérieur à

In [16]:
%%sql 
SELECT NOM, PRENOM, SALAIRE FROM EMPLOYES WHERE SALAIRE > 13000

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
5 rows affected.


[('Giroux', 'Jean-Claude', Decimal('150000.00')),
 ('Fuller', 'Andrew', Decimal('96000.00')),
 ('Brasseur', 'Hervé', Decimal('147000.00')),
 ('Leger', 'Pierre', Decimal('19000.00')),
 ('Splingart', 'Lydia', Decimal('16000.00'))]

In [17]:
%%sql 
SELECT NOM, PRENOM, DATE_EMBAUCHE 
FROM EMPLOYES 
WHERE DATE_EMBAUCHE > '2003-06-30'
FETCH NEXT 5 ROWS ONLY

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
5 rows affected.


[('Fuller', 'Andrew', datetime.date(2003, 8, 9)),
 ('Pagani', 'Hector', datetime.date(2003, 7, 2)),
 ('Chaussende', 'Maurice', datetime.date(2003, 8, 10)),
 ('Cremel', 'Brigitte', datetime.date(2003, 8, 28)),
 ('Gregoire', 'Renée', datetime.date(2003, 7, 19))]

In [18]:
%%sql 
SELECT NOM,PRENOM 
FROM EMPLOYES 
WHERE NOM > 'Twardowski' 
ORDER BY NOM

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
6 rows affected.


[('Urbaniak', 'Isabelle'),
 ('Valot', 'Alain'),
 ('Viry', 'Yvan'),
 ('Weiss', 'Sylvie'),
 ('Ziliox', 'Francoise'),
 ('Zonca', 'Virginie')]

## Inférieur à

In [19]:
%%sql 
SELECT NOM, PRENOM, DATE_EMBAUCHE 
FROM EMPLOYES 
WHERE DATE_EMBAUCHE < '1990-01-01'

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
5 rows affected.


[('Malejac', 'Yannick', datetime.date(1987, 11, 29)),
 ('Maurer', 'Véronique', datetime.date(1989, 5, 27)),
 ('Alvarez', 'Marcel', datetime.date(1988, 8, 3)),
 ('Peacock', 'Margaret', datetime.date(1989, 4, 16)),
 ('Burst', 'Jean-Yves', datetime.date(1988, 10, 23))]

In [20]:
%%sql 
SELECT NOM_PRODUIT, NO_FOURNISSEUR 
FROM PRODUITS 
WHERE NO_FOURNISSEUR <= 2
FETCH NEXT 5 ROWS ONLY

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
5 rows affected.


[('Cherry Pie Filling', Decimal('2')),
 ('Chai', Decimal('1')),
 ('Chang', Decimal('1')),
 ('Aniseed Syrup', Decimal('1')),
 ("Chef Anton's Cajun Seasoning", Decimal('2'))]

In [21]:
%%sql 
SELECT NOM_PRODUIT, QUANTITE 
FROM PRODUITS 
WHERE QUANTITE >'50'

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
5 rows affected.


[('Manjimup Dried Apples', '50 cartons (300 g)'),
 ('Thüringer Rostbratwurst', '50 sacs x 30 saucisses'),
 ('Dried Apples', '50\xa0paquets de 300\xa0g'),
 ('Amandes', 'Paquet de 5\xa0kg'),
 ('Dried Plums', 'Sac de 450\xa0g')]

In [27]:
%%sql 
SELECT CHR(80), ASCII('P') 

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
1 rows affected.


[('P', 80)]

In [28]:
%sql SELECT CODE_CLIENT, SOCIETE FROM CLIENTS WHERE CODE_CLIENT < 'B'

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
4 rows affected.


[('ALFKI', 'Alfreds Futterkiste'),
 ('ANATR', 'Ana Trujillo Emparedados y helados'),
 ('ANTON', 'Antonio Moreno Taquería'),
 ('AROUT', 'Around the Horn')]

## Différent de

In [29]:
%%sql 
SELECT NOM, PRENOM, FONCTION 
FROM EMPLOYES 
WHERE FONCTION <>'Représentant(e)'
FETCH NEXT 10 ROWS ONLY

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
10 rows affected.


[('Giroux', 'Jean-Claude', 'Président'),
 ('Fuller', 'Andrew', 'Vice-Président'),
 ('Brasseur', 'Hervé', 'Vice-Président'),
 ('Buchanan', 'Steven', 'Chef des ventes'),
 ('Leger', 'Pierre', 'Chef des ventes'),
 ('Belin', 'Chantal', 'Chef des ventes'),
 ('Chambaud', 'Axelle', 'Chef des ventes'),
 ('Ragon', 'André', 'Chef des ventes'),
 ('Splingart', 'Lydia', 'Chef des ventes'),
 ('Poupard', 'Claudette', 'Assistante commerciale')]

# L’opérateur LIKE
Il utilise deux caractères spéciaux pour signifier le type de correspondance recherchée :<br>
* un signe pourcentage « **%** », appelé caractère générique, 
* et un caractère de soulignement « **_** », appelé marqueur de position.


In [30]:
%%sql SELECT NOM_PRODUIT, QUANTITE 
FROM PRODUITS
WHERE QUANTITE LIKE '%boîtes%kg%'

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
4 rows affected.


[('Konbu', '1 boîtes (2 kg)'),
 ('Alice Mutton', '20 boîtes (1 kg)'),
 ('Filo Mix', '16 boîtes (2 kg)'),
 ('Long Grain Rice', '16\xa0boîtes de 2\xa0kg')]

Les valeurs contenues dans les colonnes sont sensibles à la case (majuscule, minuscule), les informations saisies dans les chaînes de caractères de comparaison doivent l’être aussi.

In [31]:
%%sql 
SELECT NOM_PRODUIT,QUANTITE 
FROM PRODUITS 
WHERE QUANTITE LIKE '__0%g'

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
10 rows affected.


[('Pears', '430\xa0g'),
 ('Peaches', '430\xa0g'),
 ('Pineapple', '430\xa0g'),
 ('Cherry Pie Filling', '430\xa0g'),
 ('Green Beans', '400\xa0g'),
 ('Corn', '400\xa0g'),
 ('Peas', '400\xa0g'),
 ('Tuna Fish', '140\xa0g'),
 ('Smoked Salmon', '140\xa0g'),
 ('Fruit Cocktail', '430\xa0g')]

In [32]:
%%sql 
SELECT NOM_PRODUIT, QUANTITE 
FROM PRODUITS 
WHERE QUANTITE LIKE '___%pièces' 

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
8 rows affected.


[('Teatime Chocolate Biscuits', '10 boîtes x 12 pièces'),
 ("Sir Rodney's Scones", '24 cartons x 4 pièces'),
 ('Perth Pasties', '48 pièces'),
 ('Escargots de Bourgogne', '24 pièces'),
 ('Wimmers gute Semmelknödel', '20 sacs x 4 pièces'),
 ('Scottish Longbreads', '10 sacs x 8 pièces'),
 ('Chocolate Biscuits Mix', '10\xa0boîtes de 12\xa0pièces'),
 ('Scones', '24\xa0paquets de 4\xa0pièces')]

In [35]:
%%sql 
SELECT NOM_CATEGORIE, DESCRIPTION 
FROM CATEGORIES 
WHERE DESCRIPTION LIKE '%' || NOM_CATEGORIE ||'%'

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
4 rows affected.


[('Boissons', 'Boissons, cafés, thés, bières'),
 ('Desserts', 'Desserts et friandises'),
 ('Viandes', 'Viandes préparées'),
 ('Viande en conserve', 'Viande en conserve')]

# Atelier 3.1	Les opérateurs logiques
Écrivez les requêtes vous permettant d’afficher :
>> * Le nom, le prénom et la date de naissance de tous les employés de la société.<br>
>> * Le nom de la société, la ville et le pays de tous les fournisseurs.<br>
>> * La fonction de tous les employés.<br>
>> * Toutes les fonctions des employés de l’entreprise, chaque fonction doit être affichée une seule fois.<br>
>> * La liste des localités dans lesquelles la société a au moins un client.<br>


## La restriction
Écrivez les requêtes permettant d’afficher :
>> * Le nom de la société et de la localité des clients qui habitent à **Toulouse**.<br>
>> * Le nom, le prénom et la fonction des employés qui ne sont pas des représentants.<br>
>> * Le nom du produit, la catégorie et le fournisseur des produits qui ne sont pas disponibles, le champ **INDISPONIBLE** est égal à 1.<br>
>> * Le nom, prénom et fonction des employés qui ont un salaire inférieur à **3500**.<br>
>> * Le nom, prénom et fonction des employés dirigés par l’employé numéro **86**.<br>
>> * Le nom, prénom et fonction des employés recrutés après **01/01/2003**.<br>

## Le traitement des chaînes de caractères
Écrivez les requêtes permettant d’afficher :
>> * Les produits et leur quantité conditionnée en bouteilles d’un litre.<br>

# L’opérateur BETWEEN

In [36]:
%%sql 
SELECT NOM, PRENOM, SALAIRE 
FROM EMPLOYES 
WHERE SALAIRE BETWEEN 1500 AND 2000

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
7 rows affected.


[('Poupard', 'Claudette', Decimal('1800.00')),
 ('Etienne', 'Brigitte', Decimal('2000.00')),
 ('Grangirard', 'Patricia', Decimal('1700.00')),
 ('Guerdon', 'Béatrice', Decimal('1700.00')),
 ('Devie', 'Thérèse', Decimal('1540.00')),
 ('Pouetre', 'Camille-Hélène', Decimal('2000.00')),
 ('Ziliox', 'Francoise', Decimal('1900.00'))]

In [38]:
%%sql 
SELECT NOM, PRENOM, DATE_NAISSANCE 
FROM EMPLOYES 
WHERE DATE_NAISSANCE BETWEEN '1985-01-01' AND '1985-12-31'

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
4 rows affected.


[('Chaussende', 'Maurice', datetime.date(1985, 1, 31)),
 ('Dupouey', 'Pierre', datetime.date(1985, 11, 13)),
 ('Rollet', 'Philippe', datetime.date(1985, 3, 9)),
 ('Coutou', 'Myriam', datetime.date(1985, 8, 31))]

In [39]:
%%sql 
SELECT CODE_CLIENT, SOCIETE, VILLE  
FROM CLIENTS 
WHERE CODE_CLIENT BETWEEN 'GALED' AND 'HANAR'

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
6 rows affected.


[('GALED', 'Galería del gastrónomo', 'Barcelona'),
 ('GODOS', 'Godos Cocina Típica', 'Sevilla'),
 ('GOURL', 'Gourmet Lanchonetes', 'Campinas'),
 ('GREAL', 'Great Lakes Food Market', 'Eugene'),
 ('GROSR', 'GROSELLA-Restaurante', 'Caracas'),
 ('HANAR', 'Hanari Carnes', 'Rio de Janeiro')]

# L’opérateur IN

In [40]:
%%sql 
SELECT SOCIETE, VILLE  
FROM CLIENTS 
WHERE VILLE IN ('Paris','Strasbourg','Toulouse')

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
4 rows affected.


[("La maison d'Asie", 'Toulouse'),
 ('Paris spécialités', 'Paris'),
 ('Spécialités du monde', 'Paris'),
 ('Blondel père et fils', 'Strasbourg')]

# Les expressions logiques
## AND

In [41]:
%%sql 
SELECT NOM_PRODUIT,QUANTITE, UNITES_STOCK 
FROM PRODUITS 
WHERE  UNITES_STOCK > 100 
  AND  QUANTITE LIKE '%boîte%'

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
4 rows affected.


[('Boston Crab Meat', '24 boîtes (4 onces)', Decimal('123')),
 ('Pâté chinois', '24 boîtes x 2 tartes', Decimal('115')),
 ('Crab Meat', '24\xa0boîtes de 115\xa0g', Decimal('120')),
 ('Green Tea', '20\xa0sacs par boîte', Decimal('125'))]

In [42]:
%%sql 
SELECT NO_COMMANDE, CODE_CLIENT, NO_EMPLOYE, DATE_ENVOI 
FROM COMMANDES 
WHERE CODE_CLIENT = 'COMMI' 
  AND NO_EMPLOYE BETWEEN 6 AND 8 
  AND DATE_ENVOI < '2010-04-01'

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
4 rows affected.


[(Decimal('216391'), 'COMMI', Decimal('7'), datetime.date(2010, 3, 20)),
 (Decimal('216496'), 'COMMI', Decimal('7'), datetime.date(2010, 3, 31)),
 (Decimal('216337'), 'COMMI', Decimal('7'), datetime.date(2010, 3, 21)),
 (Decimal('215408'), 'COMMI', Decimal('7'), datetime.date(2010, 2, 13))]

## OR

In [43]:
%%sql 
SELECT NOM_PRODUIT, QUANTITE, UNITES_STOCK 
FROM PRODUITS 
WHERE QUANTITE IS NULL OR UNITES_STOCK IS NULL

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
10 rows affected.


[('Hot Cereal', None, Decimal('60')),
 ('Mishi Kobe Niku', '18 cartons (500 g)', None),
 ('Alice Mutton', '20 boîtes (1 kg)', None),
 ('Rössle Sauerkraut', '25 canettes (825 g)', None),
 ('Singaporean Hokkien Fried Mee', '32 cartons (1 kg)', None),
 ('Perth Pasties', '48 pièces', None),
 ('Guaraná Fantástica', '12 canettes (355 ml)', None),
 ('Thüringer Rostbratwurst', '50 sacs x 30 saucisses', None),
 ('Granola', None, Decimal('100')),
 ('Potato Chips', None, Decimal('200'))]

In [44]:
%%sql 
SELECT NOM_PRODUIT, CODE_CATEGORIE 
FROM PRODUITS 
WHERE NO_FOURNISSEUR = 1 OR NO_FOURNISSEUR = 5

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
8 rows affected.


[('Green Beans', Decimal('9')),
 ('Peas', Decimal('9')),
 ('Chai', Decimal('1')),
 ('Chang', Decimal('1')),
 ('Aniseed Syrup', Decimal('2')),
 ('Queso Cabrales', Decimal('4')),
 ('Queso Manchego La Pastora', Decimal('4')),
 ('Walnuts', Decimal('7'))]

In [45]:
%%sql 
SELECT SOCIETE, PAYS, FAX 
FROM FOURNISSEURS 
WHERE FAX IS NULL AND PAYS = 'Allemagne' OR PAYS = 'France'

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
5 rows affected.


[('Heli Süßwaren GmbH  Co. KG', 'Allemagne', None),
 ('Plutzer Lebensmittelgroßmärkte AG', 'Allemagne', None),
 ('Aux joyeux ecclésiastiques', 'France', '01.03.83.00.62'),
 ('Escargots Nouveaux', 'France', None),
 ('Gai pâturage', 'France', '04.38.76.98.58')]

In [46]:
%%sql 
SELECT SOCIETE, PAYS, FAX 
FROM FOURNISSEURS 
WHERE FAX IS NULL AND (PAYS = 'Allemagne' OR PAYS = 'France')

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
3 rows affected.


[('Heli Süßwaren GmbH  Co. KG', 'Allemagne', None),
 ('Plutzer Lebensmittelgroßmärkte AG', 'Allemagne', None),
 ('Escargots Nouveaux', 'France', None)]

## NOT

In [47]:
%%sql 
SELECT NOM, FONCTION, SALAIRE 
FROM EMPLOYES 
WHERE SALAIRE > 2500 AND NOT FONCTION LIKE 'Rep%'

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
9 rows affected.


[('Giroux', 'Président', Decimal('150000.00')),
 ('Fuller', 'Vice-Président', Decimal('96000.00')),
 ('Brasseur', 'Vice-Président', Decimal('147000.00')),
 ('Buchanan', 'Chef des ventes', Decimal('13000.00')),
 ('Leger', 'Chef des ventes', Decimal('19000.00')),
 ('Belin', 'Chef des ventes', Decimal('10000.00')),
 ('Chambaud', 'Chef des ventes', Decimal('12000.00')),
 ('Ragon', 'Chef des ventes', Decimal('13000.00')),
 ('Splingart', 'Chef des ventes', Decimal('16000.00'))]

## NOT IN

In [48]:
%%sql 
SELECT NOM,PRENOM, TITRE  
FROM EMPLOYES 
WHERE TITRE NOT IN ('M.','Mme')

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
4 rows affected.


[('Davolio', 'Nancy', 'Mlle'),
 ('Callahan', 'Laura', 'Mlle'),
 ('Leverling', 'Janet', 'Mlle'),
 ('Dodsworth', 'Anne', 'Mlle')]

## NOT BETWEEN

In [49]:
%%sql 
SELECT NOM, PRENOM, DATE_EMBAUCHE 
FROM EMPLOYES 
WHERE DATE_EMBAUCHE NOT BETWEEN '1990-01-01' AND '2003-07-31'
ORDER BY DATE_EMBAUCHE

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
9 rows affected.


[('Malejac', 'Yannick', datetime.date(1987, 11, 29)),
 ('Alvarez', 'Marcel', datetime.date(1988, 8, 3)),
 ('Burst', 'Jean-Yves', datetime.date(1988, 10, 23)),
 ('Peacock', 'Margaret', datetime.date(1989, 4, 16)),
 ('Maurer', 'Véronique', datetime.date(1989, 5, 27)),
 ('Fuller', 'Andrew', datetime.date(2003, 8, 9)),
 ('Chaussende', 'Maurice', datetime.date(2003, 8, 10)),
 ('Cremel', 'Brigitte', datetime.date(2003, 8, 28)),
 ('Regner', 'Charles', datetime.date(2003, 9, 7))]

## IS NOT NULL

In [50]:
%%sql 
SELECT NOM, PRENOM, SALAIRE, COMMISSION 
FROM EMPLOYES 
WHERE COMMISSION IS NOT NULL AND SALAIRE > 10000

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
5 rows affected.


[('Buchanan', 'Steven', Decimal('13000.00'), Decimal('12940.00')),
 ('Leger', 'Pierre', Decimal('19000.00'), Decimal('11150.00')),
 ('Chambaud', 'Axelle', Decimal('12000.00'), Decimal('11600.00')),
 ('Ragon', 'André', Decimal('13000.00'), Decimal('5980.00')),
 ('Splingart', 'Lydia', Decimal('16000.00'), Decimal('16480.00'))]

## NOT LIKE

In [51]:
%%sql 
SELECT NOM_CATEGORIE, DESCRIPTION 
FROM CATEGORIES 
WHERE DESCRIPTION NOT LIKE '%' || NOM_CATEGORIE ||'%'

 * postgresql://stagiaire:***@51.91.76.248:15432/cours
6 rows affected.


[('Condiments', 'Sauces, assaisonnements et épices'),
 ('Produits laitiers', 'Fromages'),
 ('Pâtes et céréales', 'Pains, biscuits, pâtes et céréales'),
 ('Produits secs', 'Fruits secs, raisins, autres'),
 ('Poissons et fruits de mer', 'Poissons, fruits de mer, escargots'),
 ('Conserves', 'Fruits, légumes en conserve et confitures')]

# Atelier 3.2	Les opérateurs logiques

<img src="https://raw.githubusercontent.com/rbizoi/IntelligenceEnDonneesDeSante/main/InterrogationSQL/images/612.png" width="800">

## L’opérateur BETWEEN
Écrivez les requêtes permettant d’afficher :
>> * Le nom, prénom, fonction et salaire des employés qui ont un salaire compris entre **3500** et **6000**.<br><br>
>> * **Le numéro de commande, code client et la date de commande pour les commandes passées entre le **'01/01/2011'** et **'03/01/2011'**.<br><br>

## La comparaison avec des listes
Écrivez les requêtes permettant d’afficher :
>> * Le nom de la société, l’adresse, le téléphone et la ville des clients qui habitent à **Toulouse**, à **Strasbourg**, à **Nantes** ou à **Marseille**.<br><br>
>> * **Le nom du produit, le fournisseur, la catégorie et les quantités en stock pour les produits qui sont d’une des catégories 1, 3, 5 et 7.**<br><br>
>> * Le numéro de commande, code client et la date de commande pour les commandes passées dans une des dates : **'18/02/2011'**, **'20/02/2011'** ou **'25/02/2011'**.<br>

## L’assemblage des expressions

>> * Le nom, prénom, fonction et le salaire des représentants qui sont en activité depuis **'10/10/2002'**.<br>
>> * Le nom, prénom, fonction et le salaire des employés qui sont âgés de plus de 45 ans ou qui ont une ancienneté de plus de 10 ans.<br><br>
>> * **Le nom du produit, le fournisseur, la catégorie et les quantités des produits qui ont le numéro fournisseur entre 1 et 3 ou un code catégorie entre 1 et 3 et pour lesquelles les quantités sont données en boîtes ou en cartons.**<br><br>
>> * Les produits et leur quantité à condition que leur emballage ne soit pas d’un de ces types : cartons, boîtes ou unités et qu’il ne soit pas conditionné par paquets de 24 ou 32. Il ne faut pas afficher les produits de catégorie 1, 4 et 8.<br>